### ISSUE TITLE
make check fails on 1.0.5 with OTP 23

### ISSUE THREAD

**Issue by nolash** [davisp/jiffy#203]
```
$ make check
./rebar compile
==> jiffy (compile)
./rebar eunit skip_deps=true
==> jiffy (eunit)
======================== EUnit ========================
module 'jiffy_utf8'
undefined
=WARNING REPORT==== 4-Jul-2020::20:18:25.000027 ===
The on_load function for module jiffy returned:
{error,{load_failed,"Failed to load NIF library: '/usr/lib/erlang/lib/jiffy-1.0.5/priv/jiffy.so: cannot open shared object file: No such file or directory'"}}

*** test module not found ***
**jiffy

=======================================================
  Failed: 0.  Skipped: 0.  Passed: 0.
One or more tests were cancelled.
ERROR: One or more eunit tests failed.
ERROR: eunit failed while processing /home/lash/src/packages/arch/own/erlang-jiffy/src/jiffy: rebar_abort
make: *** [Makefile:23: eunit] Error 1
```

seems like it insists looking for the `.so` in absolute system path. I tried running `make CFLAGS -L/path/to/priv` too, but didn't help

**Comment by philipcristiano** [comment]
I ran into this as well on macos/darwin when upgrading jiffy for 23. Since older versions don't work with 23, I ran `git bisect` with 22 and it narrowed it down to feaf10f2d50e454f55a9d6b20c289031b0980503. 

When I remove the flto options this works for me on 22 and 23

**Comment by philipcristiano** [comment]
@davisp thoughts?

**Comment by cristianberneanu** [comment]
FLTO also causes issues for me on Linux Mint 19.1:
```
/usr/bin/ld: /usr/lib/llvm-9/bin/../lib/LLVMgold.so: error loading plugin: /usr/lib/llvm-9/bin/../lib/LLVMgold.so: cannot open shared object file: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
sh(c++ c_src/decoder.o c_src/encoder.o c_src/jiffy.o c_src/termstack.o c_src/utf8.o c_src/util.o c_src/doubles.o c_src/objects.o c_src/double-conversion/bignum-dtoa.o c_src/double-conversion/bignum.o c_src/double-conversion/cached-powers.o c_src/double-conversion/diy-fp.o c_src/double-conversion/double-conversion.o c_src/double-conversion/fast-dtoa.o c_src/double-conversion/fixed-dtoa.o c_src/double-conversion/strtod.o  -flto -lstdc++ -shared  -L"/home/cristian/.asdf/installs/erlang/23.1/lib/erl_interface-4.0.1/lib" -lei -o priv/jiffy.so)
failed with return code 1 and the following output:
/usr/bin/ld: /usr/lib/llvm-9/bin/../lib/LLVMgold.so: error loading plugin: /usr/lib/llvm-9/bin/../lib/LLVMgold.so: cannot open shared object file: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Removing the FLTO options fixes the problem.

**Comment by davisp** [comment]
This should have been fixed as of #186. Let me know if that's not true.
